### PR TITLE
Make it so pin list display the current revision of the pinned repo

### PR DESCRIPTION
**Pull request by desumn**
Implementation of issue #5533 

It works, but I'm not sure it's the best way to do it

### CODE REVIEW

**Comment by desumn** [comment]
In particular, may the url or revision be empty for a git/hs/darcs kind?

File: src/client/opamPinCommand.ml [hunk 1]
```diff
@@ -764,10 +764,26 @@ let list st ~short =
                   (OpamPackage.version_to_string inst))]
         with Not_found -> OpamConsole.colorise `yellow "(uninstalled)", []
       in
+      let vcs_revision = 
+        let url = OpamStd.Option.default OpamFile.URL.empty url in
+        match kind with 
+        | "git" | "hg" | "darcs" ->
```
Comment (rjbou) [inline src/client/opamPinCommand.ml:770]: You can match on `OpamUrl.backend` here. Here is an example:
https://github.com/ocaml/opam/blob/8e3ee689a9cf34aa0c6203f8d948b2d20075d9e5/src/state/opamFileTools.ml#L394-L397

File: master_changes.md [hunk 2]
```diff
@@ -39,6 +39,7 @@ users)
 ## Config
 
 ## Pin
+  * [NEW] Make it so pin list display the current revision of a pinned repository [#6274 @desumn]
```
Comment (rjbou) [inline master_changes.md:42]:
```suggestion
  * [NEW] Make it so pin list display the current revision of a pinned repository [#6274 @desumn - fix #5533]
```

**Review by rjbou — COMMENTED** [review]
Thank you for you PR!
You'll find some comments specific to the implementation the implementation.
To complete the PR, it needs also to:
* [x] add a test (we can do it)
* [ ] update the list manpage text to inform about the new column, especially that it is the hash of the current state of the source, not the installed one.
* [ ] polishing (remove trailing whitespace, indent, 80 columns), but we will do it at the end.

A todo for the dev team: we'll need to add a more exhaustive test to pin, existing one tests only the several kind of pin (vcs/path/version), but not the not synchronised package.

**Review by kit-ty-kate — CHANGES_REQUESTED** [review]

File: src/client/opamPinCommand.ml [hunk 3]
```diff
@@ -764,10 +764,26 @@ let list st ~short =
...
+          if (OpamCoreConfig.(!r.verbose_level >= 3))
+          then 
+          let rev_opt =
+            OpamProcess.Job.run 
+            @@ OpamRepository.revision
+               (OpamSwitchState.source_dir st nv)
+               (OpamFile.URL.url url) in 
+          let rev = OpamStd.Option.default OpamPackage.Version.default rev_opt in 
+          OpamPackage.Version.to_string rev
```
[the remark this answers is not in the document]
Comment (desumn) [reply src/client/opamPinCommand.ml:779]: I agree

File: src/client/opamPinCommand.ml [hunk 4]
```diff
@@ -764,10 +764,26 @@ let list st ~short =
...
+        let url = OpamStd.Option.default OpamFile.URL.empty url in
+        match kind with 
+        | "git" | "hg" | "darcs" ->
+          if (OpamCoreConfig.(!r.verbose_level >= 3))
+          then 
+          let rev_opt =
+            OpamProcess.Job.run 
+            @@ OpamRepository.revision
+               (OpamSwitchState.source_dir st nv)
+               (OpamFile.URL.url url) in
```
[the remark this answers is not in the document]
Comment (desumn) [reply src/client/opamPinCommand.ml:777]: Yes, however I don't really know where to cache the result

**Review by kit-ty-kate — APPROVED** [review]
lgtm. Thanks for this contribution!
We'll do the rest (fix the tests)

File: src/client/opamPinCommand.ml (hunk 4 above)
[the remark this answers is not in the document]
Comment (kit-ty-kate) [reply src/client/opamPinCommand.ml]: In view of https://github.com/ocaml/opam/pull/6278 and similar uses of `OpamRepository.revision` it sounds totally fine to not cache this. If we want it in the future we can do it separately.

**Comment by rjbou** [comment]
Discussed on dev meeting: We end up converging to this syntax
image

File: .github/scripts/main/main.sh [hunk 5]
```diff
@@ -127,6 +127,9 @@ if [ "$OPAM_TEST" = "1" ]; then
   fi
   cd $CACHE/opam-rt
   git fetch origin
+  if [ "$GITHUB_EVENT_NAME" = "pull_request" ] && git ls-remote --exit-code origin $GITHUB_PR_USER/$BRANCH ; then
```
Comment (kit-ty-kate) [inline .github/scripts/main/main.sh:130]:
```suggestion
  if [ "$GITHUB_EVENT_NAME" = "pull_request" ] && git ls-remote --exit-code origin "$GITHUB_PR_USER/$BRANCH" ; then
```

**Review by rjbou — APPROVED** [review]
Thanks!!

**Pull request merged**